dali/kernels/tensor_shape.h

```diff
@@ -645,6 +647,35 @@ bool is_uniform(const TensorListShape<ndim> &tls) {
   return true;
 }
 
+template <int out_dim, int in_dim>
+TensorShape<out_dim> convert_dim(const TensorShape<in_dim> &in) {
+  static_assert(out_dim == DynamicDimensions || in_dim == DynamicDimensions ||
+                in_dim == out_dim, "Incompatible number of dimensions"
+                " - must be equal or at least one side must be dynamic");
+  TensorShape<out_dim> out;
+  out.resize(in.size());
+  for (int i = 0; i < out.size(); i++)
+    out[i] = in[i];
+  return out;
+}
+
+template <int out_dim, int in_dim>
+typename std::enable_if<(out_dim != DynamicDimensions), TensorListShape<out_dim>>::type
+convert_dim(const TensorListShape<in_dim> &in) {
+  static_assert(out_dim == DynamicDimensions || in_dim == DynamicDimensions ||
+                in_dim == out_dim, "Incompatible number of dimensions"
+                " - must be equal or at least one side must be dynamic");
+  TensorListShape<out_dim> out = in.template to_static<out_dim>();
+  return out;
+}
+
+template <int out_dim, int in_dim>
+typename std::enable_if<(out_dim == DynamicDimensions), TensorListShape<out_dim>>::type
+convert_dim(const TensorListShape<in_dim> &in) {
```

Comment:
`TensorListShape<in_dim> &&in` variant?

Reply:
I see little reason to have it. In most cases, we need to copy anyway. The only scenario is that inside some template we have a no-op conversion from dynamic to dynamic - but even then, I don't think a move is likely.

Reply:
Second thought: for Tensor**List**Shape it might be useful.